src/status.ts

```diff
@@ -38,3 +38,38 @@ export enum Status {
   DATA_LOSS,
   UNAUTHENTICATED,
 }
+
+export const HttpCodeToRpcCodeMap = new Map([
+  [400, Status.INVALID_ARGUMENT],
+  [401, Status.UNAUTHENTICATED],
+  [403, Status.PERMISSION_DENIED],
+  [404, Status.NOT_FOUND],
+  [409, Status.ABORTED],
+  [416, Status.OUT_OF_RANGE],
+  [429, Status.RESOURCE_EXHAUSTED],
+  [499, Status.CANCELLED],
+  [501, Status.UNIMPLEMENTED],
+  [503, Status.UNAVAILABLE],
+  [504, Status.DEADLINE_EXCEEDED],
+]);
+
+// Maps HTTP status codes to gRPC status codes above.
+export function rpcCodeFromHttpStatusCode(httpStatusCode: number): number {
+  if (HttpCodeToRpcCodeMap.has(httpStatusCode)) {
+    return HttpCodeToRpcCodeMap.get(httpStatusCode)!;
+  }
+  // All 2xx
+  if (httpStatusCode >= 200 && httpStatusCode < 300) {
+    return Status.OK;
+  }
+  // All other 4xx
+  if (httpStatusCode >= 400 && httpStatusCode < 500) {
+    return Status.FAILED_PRECONDITION;
```

Comment:
Should we list specific exceptions like in the original PR?

https://github.com/googleapis/gax-dotnet/pull/502/files#diff-cd4cde179a23d7032ecd365e5728402f583b5c3047cc37e87de0efb718ba97a8R196-R216

I believe it makes sense to set `403 Forbidden` to `PermissionDenied`, etc.

Reply:
> I believe it makes sense to set 403 Forbidden to PermissionDenied, etc.
Yes. See line [58](https://github.com/googleapis/gax-nodejs/pull/1135/files#diff-8c75951c1a05c8cd89a6b4a97940f1e5770c830298aea9914aaae438c2c48f9bR58), check the httpcode maps first, if code is 403 it return matched `PermissionDenied`. Key is not in maps will be checked by range.